crates/sui-types/src/error.rs

```diff
@@ -638,6 +638,21 @@ impl SuiError {
             _ => false,
         }
     }
+
+    // Collapse TransactionInputObjectsErrors into a single SuiError
+    // if there's exactly one error.
+    pub fn collapse_if_single_transaction_input_error(&self) -> Option<SuiError> {
+        match self {
+            SuiError::TransactionInputObjectsErrors { errors } => {
+                if errors.len() != 1 {
+                    None
+                } else {
+                    Some(errors.get(0).unwrap().clone())
```

Comment:
It's possible that this error could also be a `SuiError::TransactionInputObjectsErrors` -- is it worth pushing down recursively, or is this only going to be used in tests where we can be sure of how deeply nested this error is going to be?

Reply:
It shouldn't be possible in practice. Is this what you mean by double wrapping above?